nova/scheduler/weights/affinity.py

```diff
@@ -49,7 +62,26 @@ def _weigh_object(self, host_state, request_spec):
         members = set(request_spec.instance_group.members)
         member_on_host = instances.intersection(members)
 
-        return len(member_on_host)
+        weight = len(member_on_host)
+        if weight == 0:
+            # get shards of the host
+            host_shard_aggrs = [aggr for aggr in host_state.aggregates
+                                if aggr.name.startswith(self._SHARD_PREFIX)]
+
+            group_hosts = set(request_spec.instance_group.hosts)
+            if group_hosts:
+                # get shards of the group
+                group_shard_aggrs = []
+                for host in group_hosts:
+                    group_shard_aggrs.append(self._get_aggregates(host))
+                group_shard_aggrs = set(group_shard_aggrs)
+
+                for shard in host_shard_aggrs:
+                    if shard in group_shard_aggrs:
+                        weight = weight + 0.5
```

Comment:
You know the `weight` is 0 here, because you checked for it. Why do you `+ 0.5` instead of assigning 0.5? That's confusing, because it looks like we add to something previously set in `weight`.

Reply:
fixed